### DISCUSSION TITLE
Best Error Handling Methodolgy with batch-list type of request

### DISCUSSION THREAD

**Discussion by axemixer** [fastapi/fastapi#10565] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

@app.post("/batch",response_model = List[Output])

def my_function(data:List[Input]):
   for id,item in enumarate(data):
      try :
            something
      except:
            raise HTTPException(status_code = 404 , detail = "Error")
```

### Description

Hello , I'm quite begginer and want to learn what is the best way of error handling when request is coming as json list ? 

When I try to use HTTPException , it stops all batch file but I want to see other correct request.  For instance ;

Batch Request : 

[
{A:1,
B:2},
{A:1,
B:null},
{A:2,
B:2},
]
If one of the elemet in request is failing , I dont see other correct request as well. 

I want to see something below : 

Response : 

[
{C : 3},
{error : You can not send null value - 404 },
{C:4}
]

### Operating System

Windows

### FastAPI Version

0.97.0

### Pydantic Version

1.10.4

### Python Version

3.9.13

**Comment by YAGregor** [comment, marked as answer]
In my opinion, this question is not about fastapi, but common web develop pattern, I suggest that just don't that, you would better to validate all the input at first, then make "something" never raise exception.

Or you must capture all the exception and handle it by yourself, just like
```python
from fastapi import FastAPI

@app.post("/batch",response_model = List[Output])

def _(index, input):
    try:
        ...  # something
        return ....
    exception HttpException as e:
        return {"exception": str(e)}

def my_function(data:List[Input]):
    return [_(id, item) for id, item in enumarate(data)]
```

**Comment by jgould22** [comment]
As the above post mentions it would be common to return something like 
```json
{
    "status": SUCCESS 
    "data":[...],
    "error": {
          "type": SOME_TYPE,
          "message": "some human readable message"
          "data": some data related to the message
     }
}
```

Don't use the base http exception for this.
